tests/hardware_tests/test_gaze_env.py

```diff
@@ -35,33 +35,90 @@ def init_config():
     return config
 
 
+def validate_pick_results(test_target_objects, test_result):
+    assert test_result is not None
+    assert len(test_result) == len(test_target_objects)
+    # Assert for all elements in test_result, "success" is True
+    print(f"Gaze Results: {test_result}")
+    for each_gaze_result in test_result:
+        assert each_gaze_result["success"] is True
+
+
+def validate_pick_feedback(feedback):
+    (status, message) = feedback
+    assert status is True
+    assert message == "Successfully picked the target object"
+
+
 def test_gaze():
-    config = init_config()
+    config = init_test_config()
 
-    test_target_objects = ["cup", "plush_lion", "plush_ball"]
+    test_target_objects = ["cup", "plush_lion", "plush_bear"]
     test_spot = Spot("GazeEnvHardwareTest")
     test_result = None
     with test_spot.get_lease(hijack=True):
         test_spot.power_robot()
-        gaze_controller = GazeController(config, test_spot)
+        gaze_controller = Pick(spot=test_spot, config=config, use_mobile_pick=False)
 
-        # Test gaze
+        # Test gaze execution and verify the result
         try:
-            test_result = gaze_controller.execute(
+            test_result = gaze_controller.execute_pick(
                 target_object_list=test_target_objects, take_user_input=True
             )
         except Exception:
             pytest.fail(
-                "Pytest raised an error while executing GazeController.execute from test_gaze_env.py"
+                "Pytest raised an error while executing static Pick.execute_pick() from atomic_skills.py"
             )
         finally:
-            test_spot.shutdown(should_dock=False)
+            validate_pick_results(
+                test_target_objects=test_target_objects, test_result=test_result
+            )
 
-    assert test_result is not None
-    assert len(test_result) == len(test_target_objects)
+        # Test mobile gaze execution and verify with feedback
+        try:
+            feedback = gaze_controller.execute(test_target_objects[0])
+        except Exception:
+            pytest.fail(
+                "Pytest raised an error while executing static Pick.execute() from atomic_skills.py"
+            )
+        finally:
+            validate_pick_feedback(feedback)
 
-    # Assert for all elements in test_result, "success" is True
-    # I changed a comment
-    print(f"Gaze Results: {test_result}")
-    for each_gaze_result in test_result:
-        assert each_gaze_result["success"] is True
+        test_spot.shutdown(should_dock=False)
+
+
+def test_mobile_gaze():
```

Comment:
curious how certain we are about mobile gaze working well? Is this a robust test or is it likely to fail sometimes?

Reply:
Mobile gaze is performing SIGNIFICANTLY better than static gaze. no doubt about that.

Great work @jimmytyyang @tusharsangam !!